python/lsst/ip/isr/linearize.py

```diff
@@ -26,226 +26,510 @@
+        rowInd, colIndOffset = kwargs['coeffs'][0:2]
+        table = kwargs['table']
+        log = kwargs['log']
 
-        self._detectorName = detector.getName()
-        self._detectorSerial = detector.getSerial()
-        self.checkLinearityType(detector)
-        ampInfoCat = detector.getAmplifiers()
-        rowIndList = []
-        colIndOffsetList = []
         numTableRows = table.shape[0]
-        for ampInfo in ampInfoCat:
-            rowInd, colIndOffset = ampInfo.getLinearityCoeffs()[0:2]
-            rowInd = int(rowInd)
-            if rowInd < 0 or rowInd >= numTableRows:
-                raise RuntimeError("Amplifier %s has rowInd=%s not in range[0, %s)" %
-                                   (ampInfo.getName(), rowInd, numTableRows))
-            rowIndList.append(int(rowInd))
-            colIndOffsetList.append(colIndOffset)
-        self._rowIndArr = np.array(rowIndList, dtype=int)
-        self._colIndOffsetArr = np.array(colIndOffsetList)
-
-    def __call__(self, image, detector, log=None):
-        """Correct for non-linearity
-
-        @param[in] image  image to be corrected (an lsst.afw.image.Image)
-        @param[in] detector  detector info about image (an lsst.afw.cameraGeom.Detector);
-                    the name, serial and number of amplifiers must match persisted data;
-                    the bbox from each amplifier is read;
-                    the linearization coefficients are ignored in favor of the persisted values
-        @param[in] log  logger (an lsst.log.Log), or None to disable logging;
-                    a warning is logged if any pixels are out of range of their lookup table
-
-        @return an lsst.pipe.base.Struct containing:
-        - numAmps number of amplifiers found
-        - numLinearized  number of amplifiers linearized (always equal to numAmps for this linearizer)
-        - numOutOfRange  number of pixels out of range of their lookup table (summed across all amps)
-
-        @throw RuntimeError if the linearity type is wrong or if the detector name, serial
-            or number of amplifiers does not match the saved data
-        """
-        self.checkDetector(detector)
-        ampInfoCat = detector.getAmplifiers()
-        numOutOfRange = 0
-        for ampInfo, rowInd, colIndOffset in zip(ampInfoCat, self._rowIndArr, self._colIndOffsetArr):
-            bbox = ampInfo.getBBox()
-            ampView = image.Factory(image, bbox)
-            tableRow = self._table[rowInd, :]
-            numOutOfRange += applyLookupTable(ampView, tableRow, colIndOffset)
+        rowInd = int(rowInd)
+        if rowInd < 0 or rowInd > numTableRows:
+            raise RuntimeError("LinearizeLookupTable rowInd=%s not in range[0, %s)" %
+                               (rowInd, numTableRows))
+
+        tableRow = table[rowInd, :]
+        numOutOfRange += applyLookupTable(image, tableRow, colIndOffset)
 
         if numOutOfRange > 0 and log is not None:
```

Comment:
You are already raising if the requested rowInd is out of range. What is this "numOutOfRange" counter referring to?

Reply:
That raise docstring is now updated.  We did not previously raise if out of range (despite the docstring), and should not now.  `numOutOfRange` denotes the number of pixels that have (imVal + c1) > len(table[c0]) (the required lookup table entry did not exist).